From fb063322b4339507157b5030978b5aeb7342e68b Mon Sep 17 00:00:00 2001
From: Varun Chandramohan <varunc@linux.vnet.ibm.com>
Date: Thu, 12 Mar 2009 13:42:07 +0530
Subject: [PATCH 04/23] Add Monitor Support For Neigh Table

This patch adds exclusive support to enable monitoring
neighbour table entries in ip command.

Signed-off-by: Varun Chandramohan <varunc@linux.vnet.ibm.com>
---
 ip/ipmonitor.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/ip/ipmonitor.c b/ip/ipmonitor.c
index df0fd91..a74730e 100644
--- a/ip/ipmonitor.c
+++ b/ip/ipmonitor.c
@@ -102,6 +102,7 @@ int do_ipmonitor(int argc, char **argv)
 	int laddr=0;
 	int lroute=0;
 	int lprefix=0;
+	int lneigh=0;
 
 	rtnl_close(&rth);
 	ipaddr_reset_filter(1);
@@ -124,6 +125,9 @@ int do_ipmonitor(int argc, char **argv)
 		} else if (matches(*argv, "prefix") == 0) {
 			lprefix=1;
 			groups = 0;
+		} else if (matches(*argv, "neigh") == 0) {
+			lneigh=1;
+			groups = 0;		
 		} else if (strcmp(*argv, "all") == 0) {
 			groups = ~RTMGRP_TC;
 		} else if (matches(*argv, "help") == 0) {
@@ -153,7 +157,9 @@ int do_ipmonitor(int argc, char **argv)
 		if (!preferred_family || preferred_family == AF_INET6)
 			groups |= nl_mgrp(RTNLGRP_IPV6_PREFIX);
 	}
-
+	if (lneigh) {
+		groups |= nl_mgrp(RTNLGRP_NEIGH);
+	}
 	if (file) {
 		FILE *fp;
 		fp = fopen(file, "r");
-- 
1.6.3.3

